packages/core/interfaces/src/index.ts

```diff
@@ -158,6 +165,60 @@ export interface UseLocale
   loading: ComputedProperty<boolean>;
 }
 
+export interface ProductGetters<PRODUCT, PRODUCT_FILTER> {
```

Comment:
maybe including the ProductType (simple, parent, etc) is a good way to recognize which product we are dealing with,
so `getType` method is maybe a solution here.

Reply:
the second thing is that many (magento, shopware for sure) ecommerce platforms have related products set on particular Product entity as one to many relation. Maybe we need to have a method like `getRelatedProducts` method as well.

Reply:
+1
PRODUCT_TYPES enum + getType sounds great!